src/v2/computeBidsAsks.ts

```diff
@@ -0,0 +1,78 @@
+import { BigNumber } from '@uniswap/sdk'
+
+function getAmountOut(
+  amountIn: BigNumber,
+  reservesIn: BigNumber,
+  reservesOut: BigNumber
+): { amountOut: BigNumber; reservesInAfter: BigNumber; reservesOutAfter: BigNumber } {
+  const amountOut = amountIn.eq(0)
+    ? new BigNumber(0)
+    : reservesOut.minus(reservesOut.multipliedBy(reservesIn).dividedBy(reservesIn.plus(amountIn.multipliedBy(0.997))))
+  return {
+    amountOut,
+    reservesInAfter: reservesIn.plus(amountIn),
+    reservesOutAfter: reservesOut.minus(amountOut)
+  }
+}
+
+function getAmountIn(
+  amountOut: BigNumber,
+  reservesIn: BigNumber,
+  reservesOut: BigNumber
+): { amountIn: BigNumber; reservesInAfter: BigNumber; reservesOutAfter: BigNumber } {
+  const amountIn = amountOut.eq(0)
+    ? new BigNumber(0)
+    : amountOut.isGreaterThanOrEqualTo(reservesOut)
+    ? new BigNumber(Infinity)
```

Comment:
we should never hit this case based on the logic in `computeBidsAsks, no? might be better to assert something to that effect